Doc/library/sqlite3.rst

```diff
@@ -829,35 +829,55 @@ Exceptions
 
 .. exception:: Warning
 
-   A subclass of :exc:`Exception`.
+   Exception raised for important warnings like data truncations while
+   inserting, etc.  It is a subclass of :exc:`Exception`.
```

Comment:
Actually currently it is raised in two cases:

1. When fail to encode the statement to the C string in `Connection.__call__()`, `Cursor.execute()` and `Cursor.executemany()`, replacing MemoryError and UnicodEncodeError. It is obviously incorrect use, and it will be fixed in #27654. In 2.7 it was used more correctly -- when the statement has wrong type (not str or unicode), so it is a bug of porting to 3.0.
2. When try to compile string containing multiple SQL statements in `Connection.__call__()`, `Cursor.execute()` and `Cursor.executemany()`. I am not sure that Warning is correct name for exception raised in this case. InterfaceError or ProgrammingError may be more appropriate.

Maybe we will deprecate it in future.

Reply:
>  I am not sure that Warning is correct name for exception raised in this case. InterfaceError or ProgrammingError may be more appropriate.

`ProgrammingError` sounds most correct.

> Maybe we will deprecate it in future.

It's part of PEP 249, so I guess it has to stay.

Possible future use: Since SQLite has no concept of `nan`, we could raise `Warning` when `nan` is converted to SQLite `NULL`.